File: code/mathematical-algorithms/Perfect_Number.cpp
// Part of Cosmos by OpenGenus Foundation

//Program to Calculate Perfect Numbers under n

#include<iostream>
#include<cmath>
using namespace std;
int main()
    {
     int n,k;
     cout<<"\nEnter the value of n:";
     cin>>n;
     cout<<"\nPerfect numbers under "<<n<<" are:\n";
     int i=1,s1=0,sum=0;

     for(k=1;k<=n;k++)
        {
           
           while(i<k)
            {
             if(k%i==0)
             s1=s1+i;
             i++;
            }

          i=1;

          if(k==s1)
          {
              cout<<k<<endl;
             
          }

          s1=0;
          
       }
       
    }

File: cosmos-master/code/mathematical-algorithms/perfect_number/perfect_number.php
<?php

/**
 * Part of Cosmos by OpenGenus Foundation
 *
 * @param int $n
 *
 * @return bool
 */
function isPerfectNumber($n)
{
    $sum = 0;
    for ($i = 1; $i < $n; $i++) {
        if ($n % $i === 0) {
            $sum += $i;
        }
    }
    return $sum === $n;
}

echo isPerfectNumber($number = 6) ? sprintf('%d is a perfect number.', $number) : sprintf('%d is not a perfect number.', $number);
